x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/CentroidCalculatorTests.java

```diff
@@ -250,7 +258,7 @@ public void testPolygonAsPoint() {
         CentroidCalculator calculator = new CentroidCalculator(polygon);
         assertThat(calculator.getX(), equalTo(GeoUtils.normalizeLon(point.getX())));
         assertThat(calculator.getY(), equalTo(GeoUtils.normalizeLat(point.getY())));
-        assertThat(calculator.sumWeight(), equalTo(1.0));
+        assertThat(calculator.sumWeight(), equalTo(3.0));
```

Comment:
a point should have a weight of 1?

Reply:
This is a bit of a fake test as we are trying to compute the centroid of a polygon which all points are the same. This is an invalid polygon and we should never get here. I think it is ok to consider in this case that the centroid is the sum of the centroid of each point. It has no practical consequences.

Reply:
OK